From 24a443a2089c2eb2f6bbeb0ea1495f5217103856 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sun, 8 Oct 2017 06:16:59 +0200
Subject: [PATCH 1/1] efi_loader: avoid NULL dereference in efi_dp_match

When calling bootefi hello twice a kernel dump occurs.

Neither bootefi hello nor bootefi selftest have an image
device patch. So do not try to dereference the NULL
value.

Fixes: 95c5553ea26 efi_loader: refactor boot device and loaded_image handling
Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_boottime.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/efi_loader/efi_boottime.c b/lib/efi_loader/efi_boottime.c
index e92ba63bf1..81e7d818fc 100644
--- a/lib/efi_loader/efi_boottime.c
+++ b/lib/efi_loader/efi_boottime.c
@@ -1164,7 +1164,8 @@ void efi_setup_loaded_image(struct efi_loaded_image *info, struct efi_object *ob
 		(void *)&efi_device_path_to_text;
 
 	info->file_path = file_path;
-	info->device_handle = efi_dp_find_obj(device_path, NULL);
+	if (device_path)
+		info->device_handle = efi_dp_find_obj(device_path, NULL);
 
 	list_add_tail(&obj->link, &efi_obj_list);
 }
-- 
2.14.1

